hangman.py

```diff
@@ -0,0 +1,270 @@
+    if letter in letters_guessed:
+        warn, tries = count(warn, tries)[0], count(warn, tries)[1]
+        # This output changes depending on if the entered symbol is "*" or a letter and how many warnings are left
+        print("Oops! You've already", *["used your hint." if letter == "*" else "guessed that letter."], "You have", warn, "warnings left" + str(*[" so you lose one guess" if warn == "no" else ""]) + ":", check)
+
+    # If the entered symbol is a "*", the program uses a hint and shows all possible matches
+    elif hints and letter == "*":
+        show_possible_matches(check)
+
+    # Then it checks if the entered symbol is a letter
+    elif letter not in string.ascii_lowercase:
+        warn, tries = count(warn, tries)[0], count(warn, tries)[1]
+        print("Oops! That is not a valid letter. You have", warn, "warnings left:", check)
+
+    # If the entered symbol is a letter and it's in the secret word, the program shows that the letter is guessed
+    elif letter in set(secret_word):
+        letters_guessed.add(letter)
+        check = get_guessed_word(secret_word, letters_guessed)
+        print("Good guess:", check)
+
+    # If the entered symbol is a letter and it isn't in the secret word, the program shows that the letter
+    # isn't guessed and user loses one try if it isn't a vowel letter or two if it is.
+    elif letter not in set(secret_word):
+        tries = vowel_check(letter, tries)
+        print("Oops! That letter is not in my word:", check)
+    letters_guessed.add(letter)
+    l = [tries, warn, secret_word, letters_guessed, letter, check]
+    return l
+
+
+def hangman(secret_word, hints = False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    tries, warn = 6, 3
+    # List "letters_guessed" is used as a container for letters which have already been guessed
+    letters_guessed = set()
+    # Variabe "check" shows the length of the secret word and which letters have already been guessed
+    check = get_guessed_word(secret_word, letters_guessed)
+    while tries != 0 and is_word_guessed(secret_word, check):
+        print("-" * 13, "\nYou have", tries, "guesses left.")
```

Comment:
Could you have different prints for different lines? It's will be easier to read (here and everywhere else)